src/ring_theory/polynomial/hermite.lean

```diff
@@ -124,5 +129,62 @@ begin
 end
 
 end coeff
+end polynomial
+
+/-! ### Lemmas about `polynomial.hermite_gauss` -/
+
+section gaussian
+
+/-- The real Gaussian function -/
+def gaussian : ℝ → ℝ := λ x, real.exp (-(x^2 / 2))
+
+lemma inv_gaussian_eq : gaussian⁻¹ = λ x, real.exp (x^2 / 2) :=
+by { ext, simp [gaussian, real.exp_neg] }
+
+lemma inv_gaussian_mul_gaussian (x : ℝ) : gaussian⁻¹ x * gaussian x = 1 :=
+inv_mul_cancel (real.exp_pos _).ne.symm
+
+lemma deriv_gaussian (x : ℝ) : deriv gaussian x = -x * gaussian x :=
+by simp [gaussian, mul_comm]
+
+lemma deriv_inv_gaussian (x : ℝ) : deriv gaussian⁻¹ x = x * gaussian⁻¹ x :=
+by simp [inv_gaussian_eq, mul_comm]
+
+lemma cont_diff_gaussian : cont_diff ℝ ⊤ gaussian :=
+((cont_diff_id.pow 2).div_const 2).neg.exp
+
+lemma hermite_gauss_succ (n : ℕ) :
+(λ x, (-1)^(n+1) * (gaussian⁻¹ x) * (deriv^[n+1] gaussian x))
+= id * (λ x, (-1)^n * (gaussian⁻¹ x) * (deriv^[n] gaussian x)) -
+  deriv (λ x, (-1)^n * (gaussian⁻¹ x) * (deriv^[n] gaussian x)) :=
+begin
+  ext,
+  simp only [function.iterate_succ', function.comp_app,
+             id.def, pi.mul_apply, pi.sub_apply, pow_succ],
+  rw [deriv_mul, deriv_const_mul, deriv_inv_gaussian],
+  ring,
+  { simp [inv_gaussian_eq] },
+  { simp [inv_gaussian_eq] },
+  { apply (cont_diff_top_iff_deriv.mp (cont_diff.iterate_deriv _ _ cont_diff_gaussian)).1 }
+end
+
+namespace polynomial
+
+lemma hermite_eq_deriv_gaussian (n : ℕ) :
+(λ (x : ℝ), aeval x (hermite n)) =
+λ x, (-1)^n * (gaussian⁻¹ x) * (deriv^[n] gaussian x) :=
+begin
+  induction n with n ih,
+  { simp [-pi.inv_apply, inv_gaussian_mul_gaussian] },
+  { rw [hermite_gauss_succ, hermite_succ, ← ih],
+    ext,
+    simp [aeval_def, eval₂_eq_eval_map] },
+end
+
+lemma hermite_eq_deriv_gaussian_apply :
+  ∀ (n : ℕ) (x : ℝ), aeval x (hermite n) =
+    (-1)^n * (gaussian⁻¹ x) * (deriv^[n] gaussian x) :=
+λ n x, congr_fun (hermite_eq_deriv_gaussian n) x
```

Comment:
May as well move `n` and `x` before the colon
```suggestion
lemma aeval_real_hermite_eq (x : ℝ)  (n : ℕ):
  aeval x (hermite n) = (-1)^n * (gaussian⁻¹ x) * (deriv^[n] gaussian x) :=
congr_fun (hermite_eq_deriv_gaussian n) x
```

Reply:
I think the name should refer to gaussian derivatives. Bear in mind that there are several equivalent definitions here for Hermite polynomials and we could just as easily have had `aeval x (hermite n) = aeval x ((λ p, X*p - p.derivative)^[n] 1)` or something.